src/vec3_linalgebra.c

```diff
@@ -29,3 +29,17 @@ float	ft_vec3_angle(t_vec3 v1, t_vec3 v2)
 	dot_product = ft_vec3_dot(v1, v2);
 	return (acos(dot_product / (ft_vec3_abs(v1) * ft_vec3_abs(v2))));
 }
+
+t_vec3	ft_vec3_hadamard(t_vec3 v1, t_vec3 v2)
+{
+	return ((t_vec3){
+		.x = v1.x * v2.x,
+		.y = v1.y * v2.y,
+		.z = v1.z * v2.z,
+	});
+}
+
+bool	ft_vec3_equal(t_vec3 v1, t_vec3 v2)
```

Comment:
This function is extremely spicy: equal comparison of floats is almost never want you want to do

Reply:
True, any suggestions? Somehow compare only to a certain decimal place?

Reply:
True, one needs to take the absolute value of the difference of the two. In case the second float is bigger than the first, the condition would not work as intended anymore. 

Took over the suggestion.